### PR TITLE
Feature/140 implement version of loading the results from the reports pitest

### PR DESCRIPTION
**Pull request by nikomall34**
This was my idea for the Ticket #140 
In order to get the coverage information I used logic from the [MutationHTMLReportListener](https://github.com/hcoles/pitest/blob/master/pitest-html-report/src/main/java/org/pitest/mutationtest/report/html/MutationHtmlReportListener.java) from pitest. I implemented a ResultListener I thought this is the easiest way to integrate it into the project's classpath (I mean here the project that is opened in IntelliJ and uses our plugin). 
The new ResultListener creates a new xml file which contains the coverage information for each class and the total of all classes. And example can be found in this PR.
Now to use the mutation ResultListener you need to publish it to MavenLocal (like it is done for the override-plugin). Then you need to put the following line `pitest("io.github.amosproj:coverage-plugin:1.0")` in the dependecies of the `build.gradle.kts`:

```
dependencies {
    pitest("io.github.amosproj:coverage-plugin:1.0")
}
```

And the outputFormat is called "report-coverage" this also has to be set in the `build.gradle.kts` file:
```
pitest {
    outputFormats.set(setOf("HTML", "XML", "report-coverage")))
}
```

Finally I adjusted the XML Parser so it is able to extract the information from the new xml file as well and store it in the correct data structures. This ResultListener would make the HTML parsers obsolete.

### CODE REVIEW

**Comment by lheimbs** [comment]
This looks like a good approach especially since I could not find a way to add something to the classpath from withing the override plugin.
One concern I have though is with adding yet another plugin. Couldn't we add this to the override plugin and avoid another plugin we'd have to publish?

**Comment by nikomall34** [comment]
I wouldn't mind. I just thought maybe other people also want to use this mutation Listener. When I looked at example mutation listeners I noticed many people uploaded their MutationListener to maven. I would only leave the MutationListener and the override plugin seperated because then the code is more oragnized.

**Comment by lheimbs** [comment]
My main concern is having to publish another thing. But I agree that in terms of code organization and also the override-plugin being a gradle plugin and the coverage part being a pitest dependency having the coverage part separate is better and integrating it into the override-plugin makes no sense.

Maybe instead of publishing it we could build and bundle it with the intellij plugin?
Otherwise this is definitely the way to go

**Comment by nikomall34** [comment]
Why exactly would this concern you?
Building and bundling it with the Intellij Plugin sounds also good. How would you do this?

**Comment by lheimbs** [comment]
It just raises the amount of work required from us and whoever takes this over down the line...

I have added a PoC in #152 that demos how it could work without publishing.

**Review by lheimbs — APPROVED** [review]

File: coverage-plugin/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: MIT
+// SPDX-FileCopyrightText: 2023
+
+plugins {
+    kotlin("jvm") version "1.9.21"
+    `java-gradle-plugin`
```
Comment (lheimbs) [inline coverage-plugin/build.gradle.kts:6]:
```suggestion
```

File: coverage-plugin/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    `maven-publish`
+    idea
+
+    id("com.gradle.plugin-publish") version "1.2.1"
```
Comment (lheimbs) [inline coverage-plugin/build.gradle.kts:10]:
```suggestion
```

File: coverage-plugin/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+            tags =
+                listOf(
+                    "pitmutationmate", "pitest", "mutation", "mutation testing", "mutation testing tool",
+                    "mutation testing framework", "mutation testing gradle plugin", "mutation testing ide plugin",
+                    "mutation testing intellij plugin", "mutation testing intellij idea plugin"
+                )
+            implementationClass = "io.github.amosproj.pitmutationmate.coverage.CoverageResultListenerFactory"
+        }
+    }
+}
```
Comment (lheimbs) [inline coverage-plugin/build.gradle.kts:53]:
```suggestion
```

**Pull request merged**